test/test_c10d.py

```diff
@@ -2112,6 +2112,46 @@ def test_fp16(self):
             any(torch.isinf(p.grad).any() for p in ddp_model.parameters())
         )
 
+    @requires_nccl()
+    @skip_if_not_multigpu
+    def test_multihead_attention_ddp_compatibility(self):
```

Comment:
Hey @pietern,

@zhangguanheng66 and I discussed offline, and we are not sure where we should put this test. It seems to me `test_c10d` should not cover domain API related tests, and existing domain API tests should not require distributed either. Do you think it makes sense to create a new file to host domain API distributed tests?

Reply:
@mrshenli @zhangguanheng66 I agree -- this is not a DDP specific test.

What was the property here that prevented usage of DDP? I don't understand the underlying issue here. Did you see the elaborate error message and the mention of `find_unused_parameters`?

Reply:
So if I add `find_unused_parameters` parameter in the unit test, the test pass.
I'm wondering if this is the best way to resolve the issue, instead of some changes on `nn.MultiheadAttention` module. Thanks. If so, I will instruct our users to do so. Thanks. @pietern @mrshenli